ndcube/extra_coords/table_coord.py

```diff
@@ -272,41 +277,51 @@ def dropped_world_dimensions(self):
 
 class QuantityTableCoordinate(BaseTableCoordinate):
     """
-    A lookup table made up of ``N`` `~astropy.units.Quantity` objects.
+    A lookup table up built on `~astropy.units.Quantity`.
 
-    This class can either be instantiated with N ND arrays (i.e. the output of
-    `numpy.meshgrid`) or N 1D arrays (i.e. the input to `numpy.meshgrid`).
+    Quantities must be 1-D but more than one can be provided to represent
+    different dimensions of an N-D coordinate.
 
-    Notes
-    -----
-    The reason for supporting both the input and output of meshgrid is that
-    meshgrid isn't called when ``mesh=True``, the "meshing" is done in the gWCS
-    layer.
+    Parameters
+    ----------
+    tables: one or more `~astropy.units.Quantity`
+        The coordinates. Must be 1 dimensionsal. If coordinate system is >1D,
+        multiple 1-D Quantities can be provided representing the different
+        dimensions
+
+    names: `str` or `list` of `str`
+        Custom names for the components of the QuantityTableCoord. If provided,
+        a name must be given for each input Quantity.
+
+    physical_types: str` or `list` of `str`
+        Physical types for the components of the QuantityTableCoord. If provided,
+        a physical type must be given for each input Quantity.
+        Physical types of the components of the SkyCoord. If provided,
+        a physical type must be given for each component.
     """
 
-    def __init__(self, *tables, mesh=False, names=None, physical_types=None):
+    def __init__(self, *tables, names=None, physical_types=None):
```

Comment:
I found having it there was confusing because it implied it wasn't redundant and made me think the object was conceptually more complicated and inhibited me in being able to manipulate it.

We could add `**kwargs`` so if `mesh` was provided it wouldn't error.